### PR TITLE
STORM-4004 - Upgrade Kafka Clients to 3.6.0

### PR DESCRIPTION
**Pull request by rzo1**
## What is the purpose of the change

- Upgrade old dependencies

## How was the change tested

- CI

### CODE REVIEW

File: DEPENDENCY-LICENSES [hunk 1]
```diff
@@ -127,7 +127,7 @@ List of third-party dependencies grouped by their license type.
         * Apache HttpCore (org.apache.httpcomponents:httpcore:4.4.16 - http://hc.apache.org/httpcomponents-core-ga)
         * Apache HttpCore NIO (org.apache.httpcomponents:httpcore-nio:4.4.5 - http://hc.apache.org/httpcomponents-core-ga)
         * Apache Ivy (org.apache.ivy:ivy:2.4.0 - http://ant.apache.org/ivy/)
-        * Apache Kafka (org.apache.kafka:kafka-clients:0.11.0.3 - http://kafka.apache.org)
+        * Apache Kafka (org.apache.kafka:kafka-clients:3.6.0 - https://kafka.apache.org)
```
Comment (avermeer) [inline DEPENDENCY-LICENSES:130]: ah yes this reference was so old !

File: external/storm-kafka-client/pom.xml [hunk 2]
```diff
@@ -110,7 +110,7 @@
         </dependency>
         <dependency>
             <groupId>org.apache.kafka</groupId>
-            <artifactId>kafka_2.11</artifactId>
+            <artifactId>kafka_2.13</artifactId>
```
Comment (avermeer) [inline external/storm-kafka-client/pom.xml:113]: I'm confused... aren't trying to get version 3.6.0 ?

File: external/storm-kafka-client/pom.xml [hunk 3]
```diff
@@ -130,7 +130,7 @@
...
```
Comment (avermeer) [inline external/storm-kafka-client/pom.xml:133]: same remark as previous, shouldn't it be 3.6.0 ?

**Review by avermeer — COMMENTED** [review]
I'm a bit confused by kafka_2.13 references vs goal to update to 3.6.0, can you clarify ?

File: external/storm-kafka-client/pom.xml (hunk 2 above)
Comment (rzo1) [reply external/storm-kafka-client/pom.xml:113]: This denotes the scala version, see https://mvnrepository.com/artifact/org.apache.kafka/kafka

File: external/storm-kafka-client/pom.xml (hunk 3 above)
Comment (avermeer) [reply external/storm-kafka-client/pom.xml:133]: Ah okay, that's confusing ;)

Comment (rzo1) [reply external/storm-kafka-client/pom.xml:133]: Yes. I also had to look twice.

**Comment by avermeer** [comment]
> > I'm a bit confused by kafka_2.13 references vs goal to update to 3.6.0, can you clarify ?
> 
> Just to write it here to avoid having to look into resolved comments: The _2.x denotes the Scala version used within Kafka.

Yes, understood !

**Review by jnioche — APPROVED** [review]

**Pull request merged**